### PR TITLE
SHIP-2420: Upgrade to node 16

### PR DESCRIPTION
**Pull request by serratus**
Task: https://starshipteam.atlassian.net/browse/SHIP-2420

Changes:
- Changed to node-16 image for CircleCI builds

TODO:
- [x] Test our build process for API

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -54,6 +54,7 @@ workflows:
           filters:
             branches:
               only:
+                - /^feature\/.*/
```
Comment (sgeisbacher) [inline .circleci/config.yml]: TODO: remove this before merging

File: node-deploy/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM circleci/node:12.16
+FROM cimg/node:16.1.0
```
Comment (sgeisbacher) [inline node-deploy/Dockerfile:1]: jfyi: i'm planning to switch to a more secure `distroless/nodejs` docker-image provided by google. but i have just checked it, they dont yet have v16 avail.

**Review by sgeisbacher — APPROVED** [review]
LGTM! but before merging plz remove the `-alpha.X` suffix and the `feature/*`-branch-deploy

File: node-deploy/Dockerfile (hunk 2 above)
Comment (serratus) [reply node-deploy/Dockerfile:1]: Understood. Let's keep that on our radar.

**Pull request merged**